SCClassLibrary/Common/Collections/Dictionary.sc

```diff
@@ -370,6 +370,25 @@ Dictionary : Set {
 		^-2
 	}
 
+	== { arg that;
+		if(that.isKindOf(this.class).not) { ^false };
+		if(that.size != this.size) { ^false };
+		that.keysValuesDo { |key, val|
+			if(this.at(key) != val) { ^false }
+		};
+		^true
+	}
+
+	hash {
+		var hash = this.class.hash;
+		this.keysValuesDo { arg key, item;
+			hash = hash bitXor: item.hash;
+			hash = hash bitXor: key.hash << 1;
```

Comment:
here and line 526 below are still a little suspect... better not to put a bit-shift last, because then the LSB will always be 0 (losing information). i would suggest here moving `<< 1` before the xor.